### PR TITLE
Add modal support to Dialog, update Modal story

### PR DESCRIPTION
**Pull request by m0neysha**
Screenshot 2023-10-19 at 3 23 42 PM

https://github.com/Arize-ai/ui-components/assets/6099607/cbcf25d0-345a-4603-bb04-d757379456a8

### CODE REVIEW

File: src/overlays/Modal.tsx [hunk 1]
```diff
@@ -49,6 +55,28 @@ const modalCSS = css`
       transform: translateX(0);
     }
   }
+
+  &.ac-modal--modal {
+    border: 1px solid ${theme.colors.gray500};
```
Comment (mikeldking) [inline src/overlays/Modal.tsx:60]:
```suggestion
    border: 1px solid var(--ac-global-color-gray-500);
```

File: src/overlays/Modal.tsx [hunk 2]
```diff
@@ -49,6 +55,28 @@ const modalCSS = css`
...
+    border-radius: 5px;
```
Comment (mikeldking) [inline src/overlays/Modal.tsx:61]:
```suggestion
    border-radius: var(--ac-global-rounding-medium);
```

**Review by eunicekokor — APPROVED** [review]

**Comment by m0neysha** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/overlays/Modal.tsx [hunk 3]
```diff
@@ -49,6 +55,28 @@ const modalCSS = css`
...
+    box-shadow: -10px 0px 30px 10px rgba(0, 0, 0, 0.1);
+    /* Start offset by the animation distance */
+    transform: translateX(500px);
```
Comment (mikeldking) [inline src/overlays/Modal.tsx:64]: I think animating in the modal makes sense but not from off screen and should come from the bottom. See https://react-spectrum.adobe.com/react-spectrum/Dialog.html

**Review by mikeldking — COMMENTED** [review]
Nice! Left some comments about globals and I think we should fix the entry animation.

File: package.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "version": "0.17.5",
+  "version": "0.17.6",
```
Comment (mikeldking) [inline package.json:2]:
```suggestion
  "version": "0.17.6-0",
```
Best to publish as a pre-release for testing

File: src/overlays/Modal.tsx (hunk 2 above)
Comment (m0neysha) [reply src/overlays/Modal.tsx:61]: oh nice! thanks!

File: src/overlays/Modal.tsx (hunk 3 above)
Comment (m0neysha) [reply src/overlays/Modal.tsx]: Makes sense! Fixed it to match Dialog here

**Review by mikeldking — APPROVED** [review]

**Pull request merged**